src/components/app/index.tsx

```diff
@@ -2,56 +2,176 @@ import { h, Component } from 'preact';
 import { bind, bitmapToImageData } from '../../lib/util';
 import * as style from './style.scss';
 import Output from '../output';
+import Options from '../options';
 
-import {MozJpegEncoder} from '../../lib/codec-wrappers/mozjpeg-enc';
+import { Encoder } from '../../codecs/codec';
+import IdentityEncoder from '../../codecs/identity/encoder';
+import MozJpegEncoder from '../../codecs/mozjpeg/encoder.worker';
+
+export type ImageType = 'original' | 'MozJpeg';
+
+export type CodecOptions = any;
+
+const ENCODER_NAMES = {
```

Comment:
Add a comment where this string is used. Also, can we merge `ENCODER_NAMES` and `ENCODERS` or do they need to be separate?

Reply:
And/or declare a type (especially if the two merge)